pkg/profilecreator/profilecreator.go

```diff
@@ -231,3 +233,64 @@ func (ghwHandler GHWHandler) CPU() (*cpu.Info, error) {
 func (ghwHandler GHWHandler) Topology() (*topology.Info, error) {
 	return ghw.Topology(ghwHandler.snapShotOptions)
 }
+
+// GetReservedAndIsolatedCPUs returns Reserved and Isolated CPUs
+func (ghwHandler GHWHandler) GetReservedAndIsolatedCPUs(reservedCPUCount int, splitReservedCPUsAcrossNUMA bool) (string, string, error) {
+	if reservedCPUCount < 0 {
+		return "", "", fmt.Errorf("Specified eservered CPU count is negative, please specify it correctly")
+	}
+	topologyInfo, err := ghwHandler.Topology()
+	if err != nil {
+		return "", "", fmt.Errorf("Error obtaining Topology Info from GHW snapshot: %v", err)
+	}
+	reservedCPUSet := cpuset.NewBuilder()
+	isolatedCPUSet := cpuset.NewBuilder()
+	var numaNodeNum int
+	if splitReservedCPUsAcrossNUMA {
+		numaNodeNum = len(topologyInfo.Nodes)
+	} else {
+		numaNodeNum = 1
+	}
+
+	var max = 0
+	reservedPerNuma := reservedCPUCount / numaNodeNum
+	remainder := reservedCPUCount % numaNodeNum
```

Comment:
Can't we error out if we have remainder != 0 ? That means we can't equally divide the reserved CPUs between the NUMA nodes, so we can't accomplish the task. I will add this check to the validation.

Reply:
So the what I have introduced in the code is even if the reserved CPUs cannot be divided between the NUMA nodes they can still be spread as much as possible. E.g. In case a system has 4 NUMA Nodes and the user wants 15 reserved CPUs, we would get 4 cpus from 3 NUMA nodes and 3 from 1 NUMA Node. The user might not know how many NUMA nodes are on the system and could request an arbitrary number that is not divisible by the number of NUMA nodes and code would be gracefully handle that case without returning an error.

Reply:
I am usually against the "best-effor": approach. If the user will select a "wrong" number we will error out and explain. Otherwise the user MIGHT think he reserved enough CPUs per NUMA node while he didn't...

Reply:
Okay, will update as per your suggestion.